### ISSUE TITLE
shufflenetV2 eval error

### ISSUE THREAD

**Issue by an1018** [tensorpack/tensorpack#1185]
If you're asking about an unexpected problem which you do not know the root cause,
use this template. __PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

If you already know the root cause to your problem,
feel free to delete everything in this template.

### 1. What you did:
In ImageNetModels directory,type the following command:
$python shufflenet.py --eval --data ./data --load ShuffleNetV2-0.5x.npz --v2 -r=0.5

### 2. What you observed:
got the following error:
![image](https://user-images.githubusercontent.com/19808900/57613459-7c7cfe80-75a9-11e9-9654-78bbcc674bb3.png)

### 3. What you expected, if not obvious.
It comes to nothing when i eval the model using sevel pictures

### 4. Your environment:
  python：3.6
 tensorpack：0.9.4
TensorFlow：1.8.0

**Comment by ppwwyyxx** [comment]
Either you have added bugs to the code or you're not using latest version of the code.
The command you gave will not print logs like that.

**Comment by an1018** [comment]
@ppwwyyxx Thanks for your reply!
I've tried again without change any code, still the same logs.
The difference is that I only used 10 validation images, and val images and val.txt like below:
![image](https://user-images.githubusercontent.com/19808900/57665096-379aab80-762d-11e9-92d0-742197b62737.png)
![image](https://user-images.githubusercontent.com/19808900/57665097-38334200-762d-11e9-92e1-8f72db7dd2c5.png)
Does it matter?

**Comment by ppwwyyxx** [comment]
You should have filled the issue template which will make your question more likely to be correctly answered. 

If you use a small amount of images, you'll see the error about buffer_size as you commented at https://github.com/tensorpack/tensorpack/issues/945#issuecomment-491750878. The solution is to use a small buffer_size at https://github.com/tensorpack/tensorpack/blob/aa1f82f7f1d9dded48b3bed08c2e40d998ba7fbd/examples/ImageNetModels/imagenet_utils.py#L102.

**Comment by an1018** [comment]
@ppwwyyxx Thank you very much,I see!
Could I ask you anther question?
1. What you did:
 I download the files of ilsvrc_metadata
![image](https://user-images.githubusercontent.com/19808900/57690735-e8786900-7674-11e9-994f-2ffd230f9af0.png)

2. What you expected
Q1:
   If I  train shufflenetV2 with my own dataset, do I only need to provide my own train.txt 、val.txt and test.txt？
Q2:
And train folder should like this?
![image](https://user-images.githubusercontent.com/19808900/57690992-9d128a80-7675-11e9-9543-22903518d4dd.png)

**Comment by ppwwyyxx** [comment]
Theoretically, you can modify the txt files ( `test.txt` is not used) and the image files to use your own dataset. The structure of the train folder in your image is correct.

But it is better to [write a dataflow](https://tensorpack.readthedocs.io/tutorial/extend/dataflow.html) that loads your images.

**Comment by an1018** [comment]
@ppwwyyxx You mean than I need to write a dataflow which replace dataset.ILSVRC12 function?
![image](https://user-images.githubusercontent.com/19808900/57901345-42587900-7897-11e9-9439-263bf1618edc.png)

**Comment by ppwwyyxx** [comment]
Yes.

**Comment by an1018** [comment]
@ppwwyyxx 
Oh,i see.
But  I still have two questions about train and eval:
Q1:about train
    If I train shuffleNetV2 with my own dataset, could I  use the code in shufflenet.py :
  
![image](https://user-images.githubusercontent.com/19808900/57911797-5615d680-78bb-11e9-9290-27249362c286.png)

Q2:about val
   If i eval myself model, could I  use the code in shufflenet.py  :
  
![image](https://user-images.githubusercontent.com/19808900/57911976-c1f83f00-78bb-11e9-8def-52dfe75e878e.png)

**Comment by ppwwyyxx** [comment]
If you need to change the dataset, all you need to modify is `get_data` and the number of classes in the model.

**Comment by an1018** [comment]
Thank you very much ,I will have a try!

**Comment by an1018** [comment]
@ppwwyyxx 
1. What you did:
In ImageNetModels directory, train shufflenetV2 model:
$python shufflenet.py  --data ./data --v2 -r=0.5

(The picture size I used is 32*32 , and  I have changed 224 to 32 in shufflenetv2.py and imagenet_utils.py)

2. What you observed:
got the following error:
![image](https://user-images.githubusercontent.com/19808900/57922540-93d22980-78d2-11e9-9f0e-8b4752deea63.png)

3.What you expected
Where else do I need to change the code?

**Comment by an1018** [comment]
@ppwwyyxx Sorry to trouble you again! 
1.What you did:
    I train shuffleNetV2 with my own dataset, which contains 90,000 images for training the model
   And I don't use data aug
![image](https://user-images.githubusercontent.com/19808900/57970182-f00b7b00-79b0-11e9-9fe9-8ba5ddd72560.png)
2.What you expected
  I have trained almost 500 epoches, top*-errors are still high. 
 Is it because I train less, or  training data less, or something else?
![image](https://user-images.githubusercontent.com/19808900/57970168-be92af80-79b0-11e9-8ac3-e4b767f6ab34.png)

**Comment by an1018** [comment]
@ppwwyyxx 
And I have anther question
1.What you did:
  $python shufflenet.py --eval --data ./data --load ShuffleNetV2-0.5x.npz --v2 -r=0.5
2.What you expected
  Now I can only get top-1 and top-5 error. 
  How can I get the recognition of val data?

**Comment by ppwwyyxx** [comment]
As said in the issue template:

> If you expect certain accuracy, only in one of the two conditions can we help with it: (1) You're unable to reproduce the accuracy documented in tensorpack examples. (2) It appears to be a tensorpack bug.
Otherwise, how to train a model to certain accuracy is a machine learning question. We do not answer machine learning questions and it is your responsibility to figure out how to make your models more accurate.

**Comment by ppwwyyxx** [comment]
> How can I get the recognition of val data?

https://github.com/tensorpack/tensorpack/blob/master/examples/ImageNetModels/imagenet_utils.py#L276-L284
You can change the output name to any tensor defined in the graph and it will be predicted below.

**Comment by an1018** [comment]
@ppwwyyxx Where is the build grahp function?I can't find it

**Comment by ppwwyyxx** [comment]
https://github.com/tensorpack/tensorpack/blob/8879637399b665bfd11aa5ded663619a1a19bdb9/examples/ImageNetModels/imagenet_utils.py#L334

**Comment by an1018** [comment]
@ppwwyyxx 
When I eval on my own data, the following error appears, I don't know why
![image](https://user-images.githubusercontent.com/19808900/58033282-ca1fdb00-7b56-11e9-9aa0-3e2d6213c295.png)

**Comment by ppwwyyxx** [comment]
I did not know what you did so I could not reproduce your issue. Please make sure you are calling https://tensorpack.readthedocs.io/modules/dataflow.html#tensorpack.dataflow.MultiThreadMapData with correct arguments following its documentation.
